sopel/modules/remind.py

```diff
@@ -226,47 +215,219 @@ def remind_in(bot, trigger):
         duration = int(duration)
     timezone = get_timezone(
         bot.db, bot.config, None, trigger.nick, trigger.sender)
-    create_reminder(bot, trigger, duration, reminder, timezone)
+    timestamp = create_reminder(bot, trigger, duration, reminder)
 
+    if duration >= 60:
+        human_time = format_time(
+            bot.db,
+            bot.config,
+            timezone,
+            trigger.nick,
+            trigger.sender,
+            datetime.utcfromtimestamp(timestamp))
+        bot.reply('Okay, will remind at %s' % human_time)
+    else:
+        bot.reply('Okay, will remind in %s secs' % duration)
 
-REGEX_AT = re.compile(r'(\d+):(\d+)(?::(\d+))?([^\s\d]+)? (.*)')
+
+REGEX_AT = re.compile(
+    # hours:minutes
+    r'(?P<hours>\d+):(?P<minutes>\d+)'
+    # optional seconds
+    r'(?::(?P<seconds>\d+))?'
+    # optional timezone
+    r'(?P<tz>[^\s\d]+)?'
+    # optional date (start)
+    r'(?:\s+'
+    # - date 1 (at least one digit)
+    r'(?P<date1>\d{1,4})'
+    # - separator (one character)
+    r'(?P<sep>[./-])'
+    # - date 2 (at least one digit)
+    r'(?P<date2>\d{1,4})'
+    # - optional sep + date 3 (at least one digit)
+    r'(?:(?P=sep)(?P<date3>\d{1,4}))?'
+    r')?'  # (end)
+    # at least one space + message
+    r'\s+(?P<message>.*)'
+)
+
+
+class TimeReminder(object):
+    """Time reminder for the ``at`` command"""
+    def __init__(self,
+                 hour,
+                 minute,
+                 second,
+                 timezone,
+                 date1,
+                 date2,
+                 date3,
+                 message):
+        self.hour = hour
+        self.minute = minute
+        self.second = second
+        self.timezone = pytz.timezone(timezone)
+        self.date1 = date1
+        self.date2 = date2
+        self.date3 = date3
+        self.message = message
+
+    def __eq__(self, other):
+        return all(
+            getattr(self, attr) == getattr(other, attr, None)
+            for attr in [
+                'hour',
+                'minute',
+                'second',
+                'timezone',
+                'date1',
+                'date2',
+                'date3',
+                'message',
+            ]
+        )
+
+    def get_duration(self, today=None):
+        """Get the duration between the reminder and ``today``
+
+        :param today: aware datetime to compare to, default to UTC now
```

Comment:
"aware"?

Reply:
Yup, by default, a `datetime.datetime` object are called "naive" datetime, ie. they don't know their timezone. An aware datetime object knows about its timezone, represented by their `tzinfo` property. I discovered that while working on Django, and now it's in the [Python 3 documentation](https://docs.python.org/3/library/datetime.html):

> There are two kinds of date and time objects: “naive” and “aware”.

Reply:
Would it, ahem, "fuck shit up", if "aware" was part of the `:type:` instead of `:param:`?

Reply:
That's a real good question. I need to check that.